records_mover/db/postgres/copy_options/date_input_style.py

```diff
@@ -40,342 +31,377 @@ def upgrade_date_order_style(style: DateOrderStyle, hint_name: str) -> None:
+            quiet_remove(self.unhandled_hints, 'datetimeformattz')
+        elif datetimeformattz == 'YYYY-MM-DD HH:MI:SS':
+            #
+            #
+            #  postgres=# select timestamptz '2020-01-01 23:01:01';
+            #        timestamptz
+            #  ------------------------
+            #   2020-01-01 23:01:01+00
+            #  (1 row)
+            #
+            #  postgres=#
 
-    # timeonlyformat: Valid values: "HH12:MI AM" (e.g., "1:00 PM"),
-    # "HH24:MI:SS" (e.g., "13:00:00")
+            # Any DateStyle will do as this is unambiguous
+            quiet_remove(self.unhandled_hints, 'datetimeformattz')
+        elif datetimeformattz == "MM/DD/YY HH24:MI":
+            # "MM/DD/YY HH24:MI"
+            #
+            #  postgres=# select timestamptz '01/02/2999 23:01';
+            #        timestamptz
+            #  ------------------------
+            #   2999-01-02 23:01:00+00
+            #  (1 row)
+            #
+            #  postgres=#
+            self.upgrade_date_order_style('MDY', 'datetimeformattz')
+        elif datetimeformattz == 'DD-MM-YY HH:MI:SSOF':
+            #  postgres=# select timestamptz '02-01-2999 23:01:01-10';
+            #        timestamptz
+            #  ------------------------
+            #   2999-02-02 09:01:01+00
+            #  (1 row)
+            #
+            self.upgrade_date_order_style('DMY', 'datetimeformattz')
+        elif datetimeformattz == 'DD/MM/YY HH:MI:SSOF':
+            #  postgres=# select timestamptz '02/01/99 23:01:01-10';
+            #        timestamptz
+            #  ------------------------
+            #   1999-02-02 09:01:01+00
+            #  (1 row)
+            #
+            self.upgrade_date_order_style('DMY', 'datetimeformattz')
+        elif datetimeformattz == 'DD-MM-YYYY HH:MI:SSOF':
+            #  postgres=# select timestamptz '02-01-1999 23:01:01-10';
+            #        timestamptz
+            #  ------------------------
+            #   1999-02-02 09:01:01+00
+            #  (1 row)
+            #
+            self.upgrade_date_order_style('DMY', 'datetimeformattz')
+        elif datetimeformattz == 'MM/DD/YY HH:MI:SSOF':
+            #  postgres=# select timestamptz '01/02/99 23:01:01-10';
+            #        timestamptz
+            #  ------------------------
+            #   1999-01-03 09:01:01+00
+            #  (1 row)
+            #
+            self.upgrade_date_order_style('MDY', 'datetimeformattz')
+        elif datetimeformattz == 'MM-DD-YYYY HH:MI:SSOF':
+            #  postgres=# select timestamptz '01-02-1999 23:01:01-10';
+            #        timestamptz
+            #  ------------------------
+            #   1999-01-03 09:01:01+00
+            #  (1 row)
+            #
+            self.upgrade_date_order_style('MDY', 'datetimeformattz')
+        else:
+            cant_handle_hint(self.fail_if_cant_handle_hint,
+                             'datetimeformattz', self.hints)
```

Comment:
@Brunope might it make more sense to use a dictionary for this, just having the dictionary pass back the `'MDY', 'datetimeformattz'` (or similar) as a tuple?

Reply:
Sounds good!